src/main/java/com/aws/greengrass/logmanager/model/LogFile.java

```diff
@@ -0,0 +1,100 @@
+package com.aws.greengrass.logmanager.model;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.logmanager.LogManagerService;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.aws.greengrass.util.Digest.calculate;
+
+public class LogFile extends File {
+    // custom serialVersionUID for class extends Serializable class
+    private static final long serialVersionUID = 123;
+    private static final Logger logger = LogManager.getLogger(LogManagerService.class);
+    private static final int linesNeeded = 1;
+
+    public LogFile(String pathname) {
+        super(pathname);
+    }
+
+    /**
+     * Convert the file to LogFile.
+     * @param file The file to be converted.
+     * @return
+     */
+    public static LogFile of(File file) {
+        return new LogFile(file.getAbsolutePath());
+    }
+
+    /**
+     * Convert list of files to list of LogFiles.
+     * @param files The list of files to be converted.
+     * @return
+     */
+    public static LogFile[] of(File... files) {
+        if (files == null) {
+            return new LogFile[] {};
+        }
+        return Arrays.stream(files).map(LogFile::of).toArray(LogFile[]::new);
+    }
+
+    public boolean isLogFile() {
+        return super.isFile();
+    }
```

Comment:
this name is confusing and not accurate. this method should just be called `isFile` if that's what it is doing. `isLogFile` sounds like it should only return true when it is a log file, this returns true if it is any file.